From 7142bbeb17e88f6d0895dc11fdd14c760959e9c9 Mon Sep 17 00:00:00 2001
From: Yongxin Liu <yongxin.liu@windriver.com>
Date: Wed, 20 Mar 2024 09:49:39 +0000
Subject: [PATCH] rdk/grr: remove THIS_MODULE from class_create()

THIS_MODULE was removed in kernel commit 1aaba11da9aa
("driver core: class: remove module * from class_create()").

Upstream-Status: Inappropriate [RDK is released in tarball only]

Signed-off-by: Yongxin Liu <yongxin.liu@windriver.com>
---
 klm/cpk-ae/ice_sw_ae_main.c | 2 +-
 klm/i3c/i3c-rdk.c           | 2 +-
 klm/ies/ies_main.c          | 2 +-
 klm/oobmsm/oobmsm-rdk.c     | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/klm/cpk-ae/ice_sw_ae_main.c b/klm/cpk-ae/ice_sw_ae_main.c
index c5988be..aa6cad9 100644
--- a/klm/cpk-ae/ice_sw_ae_main.c
+++ b/klm/cpk-ae/ice_sw_ae_main.c
@@ -3250,7 +3250,7 @@ skip_sim_check:
 	}
 
 	/* Create class */
-	ae_class = class_create(THIS_MODULE, "ice_sw_app_ext");
+	ae_class = class_create("ice_sw_app_ext");
 
 	if (!ae_class) {
 		pr_err("%s: class_create() returned %ld\n",
diff --git a/klm/i3c/i3c-rdk.c b/klm/i3c/i3c-rdk.c
index 1420dca..aaf429d 100644
--- a/klm/i3c/i3c-rdk.c
+++ b/klm/i3c/i3c-rdk.c
@@ -2357,7 +2357,7 @@ static int i3c_rdk_init_char(struct i3c_rdk_device *i3c_rdk_dev)
 		return rc;
 	}
 
-	i3c_rdk_dev->class = class_create(THIS_MODULE, "i3c-rdk");
+	i3c_rdk_dev->class = class_create("i3c-rdk");
 
 	if (!i3c_rdk_dev->class) {
 		pr_err("%s: class_create() failed\n", name);
diff --git a/klm/ies/ies_main.c b/klm/ies/ies_main.c
index 618631f..d557fda 100644
--- a/klm/ies/ies_main.c
+++ b/klm/ies/ies_main.c
@@ -1217,7 +1217,7 @@ static int ies_char_dev_init(void)
 	ies_dev_alloc = 0;
 	ies_major = MAJOR(dev);
 
-	ies_class = class_create(THIS_MODULE, IES_DEVICE_NAME);
+	ies_class = class_create(IES_DEVICE_NAME);
 	if (IS_ERR(ies_class)) {
 		err = PTR_ERR(ies_class);
 		ies_class = NULL;
diff --git a/klm/oobmsm/oobmsm-rdk.c b/klm/oobmsm/oobmsm-rdk.c
index 72757c1..92bc306 100644
--- a/klm/oobmsm/oobmsm-rdk.c
+++ b/klm/oobmsm/oobmsm-rdk.c
@@ -519,7 +519,7 @@ static int oobmsm_create_char_device(struct oobmsm_device *oobmsm_device)
 		return rc;
 	}
 
-	oobmsm_device->class = class_create(THIS_MODULE, DRV_NAME);
+	oobmsm_device->class = class_create(DRV_NAME);
 
 	if (!oobmsm_device->class) {
 		dev_err(dev, "%s: class_create() failed\n", DRV_NAME);
-- 
2.39.3

